acme_diags/derivations/acme.py

```diff
@@ -361,12 +375,16 @@ def cosp_bin_sum(cld, prs_low0, prs_high0, tau_low0, tau_high0):  # noqa
             )  # MODIS obs
 
     if cld.id == "CLD_MISR":  # MISR model
+        if max(prs) > 1000:  # COSP v2 cosp_htmisr in units m instead of km as in v1
+            cld = cld[
+                1:, :, :, :
+            ]  # COSP v2 cosp_htmisr[0] equals to 0 instead of -99 as in v1, therefore cld needs to be masked manually
         cld_bin = cld(cosp_htmisr=(prs_low, prs_high), cosp_tau=(tau_low, tau_high))
-        if prs_low == 7:
+        if prs_low in [7, 7000]:
             prs_lim = "high cloud fraction"
-        if prs_high == 7 and prs_low == 3:
+        if prs_high in [7, 7000] and prs_low in [3, 3000]:
             prs_lim = "middle cloud fraction"
-        if prs_high == 3:
+        if prs_high in [3, 3000]:
             prs_lim = "low cloud fraction"
```

Comment:
There is some duplicated logic to calculate `prs_lim`, which involves `prs_low` and `prs_high` variables and some bounds. We should extract it into a separate function and use tuples for the bounds.

Example
---
```python
from typing import Tuple

def determine_fraction(
    prs_low: int, prs_high: int, low_bnd: Tuple[int, int], high_bnd: Tuple[int, int]
) -> str:
    """Determines the fraction type based on prs values and the specified boundaries"""
    if prs_low in low_bnd:
        return "high cloud fraction"
    elif prs_low in low_bnd and prs_high in high_bnd:
        return "middle cloud fraction"
    elif prs_high in high_bnd:
        return "low cloud fraction"
    else:
        raise ValueError(
            "Cloud fraction could not be determined, please check arg values"
        )
```
Usage
---
```python
# Before
if prs_low in [440, 44000]:
    prs_lim = "high cloud fraction"
if prs_low in [680, 68000] and prs_high in [440, 44000]:
    prs_lim = "middle cloud fraction"
if prs_high in [680, 68000]:
    prs_lim = "low cloud fraction"

# After
prs_lim = determine_fraction(prs_low, prs_high, (440, 44000), (680, 68000))
```

Reply:
I adopted this change.